src/EnergyPlus/WaterThermalTanks.cc

```diff
@@ -7422,14 +7412,11 @@ namespace WaterThermalTanks {
 
 		// Select mode of operation (float mode or heat mode) from last iteration.
 		// Determine if heating will occur this iteration and get an estimate of the PLR
-		switch (HeatPump.Mode) {
-		case HeatMode:
+		if ( HeatPump.Mode == HeatMode ) {
```

Comment:
The `switch` statement was giving me grief, so I just went simple and changed it back to an if.